packages/interface/src/contexts/Maci.tsx

```diff
@@ -324,12 +325,17 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
 
   /// check the poll data and tally data
   useEffect(() => {
-    if (!signer) {
+    if (!window.ethereum) {
       return;
     }
 
     setIsLoading(true);
 
+    const provider = new BrowserProvider(window.ethereum as unknown as EIP1193Provider, {
```

Comment:
what benefit does this bring? and what differences are there?

Reply:
so the problem is, when we wanna load `VotingEndsAt`, we pass `signer` inside, which requires the user to connect to their wallet. However, I think we should allow users to check the current app state no matter if they connect their wallet or not.
(so currently, if user didn't connect their wallet, the `VotingEndsAt` is not loaded, so it would always be tallying, or some state other than Voting even though is under Voting period.)

Reply:
oh I see, good job with this! one more question, if you don't have a wallet at all installed then this would be undefined right?